core/src/main/java/io/grpc/GrpcCallContext.java

```diff
@@ -0,0 +1,87 @@
+package io.grpc;
+
+/**
+ * Holds state pertaining to a single GRPC call.
+ */
+public class GrpcCallContext {
```

Comment:
I like how this is its own separate concept. But I don't think we're ready for a "one context to rule them all" API, at this point though. We had something like that internally, and it grew into a horribly tightly-coupled mess as everything wanted their information stored there as a field.

We need a stronger Context API. [Go's Context API](https://godoc.org/golang.org/x/net/context) was _very_ helpful for the Go implementation. Having it be a Map and loosely coupled with its contents is probably necessary. I have open https://github.com/grpc/grpc-java/issues/262 for it.

But that is to say, that the Context part of this API is going to be the hardest part to get into a shape where I think we would merge it. For a large part because _we_ don't know what we want from the Context API yet.